src/libsyntax/attr.rs

```diff
@@ -317,6 +317,13 @@ pub fn find_export_name_attr(diag: &Handler, attrs: &[Attribute]) -> Option<Inte
     })
 }
 
+pub fn contains_extern_indicator(attrs: &[Attribute]) -> bool {
+    contains_name(attrs, "no_mangle") ||
+        contains_name(attrs, "link_section") ||
+        contains_name(attrs, "linkage") ||
+        contains_name(attrs, "export_name")
```

Comment:
The problem is the diagnostic parameter, and that this has to go through the `CrateStore::is_extern_item` trait. Constants should be used instead of string literals, but that's a change to be made to the entire compiler. I don't really feel that `find_export_name_attr` is the right way to go about this. I tried to thread it through and failed, maybe you can advise where I can get an instance of it in each of the two places it's needed?